File: Data_Structures/Arrays/binarySearch.js
let binarySearch = function(nums, target) {
    if (nums.length === 0) {
        return -1; 
    }

    let start = 0;
    let end = nums.length-1;

    while (start <= end) {
        let mid = Math.floor((end-start)/2) + start;
        if (nums[mid] === target) {
            return mid;
        }

        if (nums[mid] < target) {
            start = mid + 1;
        } else {
            end = mid - 1;
        }
    }
    return -1;
};

File: leetcode/BinarySearch.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int start = 0; 
        int end = nums.Length-1;

        int mid; 
        
        while (start <= end)
        {
            mid = start + ((end-start)/2);
            
            if (nums[mid] == target)
            {
                return mid;
            }
            
            if (target < nums[mid])
            {
                end -= 1;
            } else {
                start += 1;
            }
        }
        return -1;
    }
}